web/impact/impact/tests/test_algolia_api_key_view.py

```diff
@@ -53,10 +81,24 @@ def test_unauthenticated_user_is_denied(self):
                 response_data['detail'] == 'Authentication credentials '
                                            'were not provided.')
 
-    def test_staff_user_gets_empty_filters(self):
+    def test_entrepreneur_that_never_had_a_finalist_role_gets_403(self):
+        user = self._create_entrepreneur()
+        with self.settings(
+                ALGOLIA_APPLICATION_ID='test',
+                ALGOLIA_API_KEY='test'):
+            with self.login(email=user.email):
+                response = self.client.get(self.url)
+                self.assertTrue(response.status_code, 403)
+
+    def test_user_with_staff_role_grant_sees_all_mentors(self):
         user = self.basic_user()
-        user.is_staff = True
-        user.save()
+
+        named_group = NamedGroupFactory()
```

Comment:
The fact that you're adding the program with an explicit `mentor_program_group` suggests that the `mentor_program_group` is important, but it's not clear to me how since it doesn't show up in your code changes. Can you clarify? What happens if it's not there, and is there a test about that?

Reply:
there's this [logic](https://github.com/masschallenge/impact-api/blob/95f528908b7c4775f517eac2a0dc2a3e20b93802/web/impact/impact/views/algolia_api_key_view.py#L88) where we do this

```
program_groups = Program.objects.filter(
        programrole__in=user_program_roles_as_participant).values_list(
        'mentor_program_group', flat=True).distinct()
    program_families = ProgramFamily.objects.filter(
        programs__mentor_program_group__in=program_groups).prefetch_related(
        'programs').distinct()
    facet_filters = _facet_filters(program_families)
```

I've been meaning to get some context from Shanks about `_facet_filters` but it looks like `mentor_program_group` is used in a grander idea to get the most recent program of a program family to grant the user access to that programs mentors.